src/EventStore.Client.Streams/EventStoreClient.Append.cs

```diff
@@ -92,105 +92,136 @@ public partial class EventStoreClient {
+				CancellationToken cancellationToken
+				) {
+			using var call = new Streams.Streams.StreamsClient(callInvoker).Append(
+					EventStoreCallOptions.CreateNonStreaming(Settings, deadline, userCredentials, cancellationToken)
+					);
 
 			IWriteResult writeResult;
+
 			try {
 				await call.RequestStream.WriteAsync(header).ConfigureAwait(false);
 
 				foreach (var e in eventData) {
-					_log.LogTrace("Appending event to stream - {streamName}@{eventId} {eventType}.",
-						header.Options.StreamIdentifier, e.EventId, e.Type);
-					await call.RequestStream.WriteAsync(new AppendReq {
-						ProposedMessage = new AppendReq.Types.ProposedMessage {
-							Id             = e.EventId.ToDto(),
-							Data           = ByteString.CopyFrom(e.Data.Span),
-							CustomMetadata = ByteString.CopyFrom(e.Metadata.Span),
-							Metadata = {
-								{Constants.Metadata.Type, e.Type},
-								{Constants.Metadata.ContentType, e.ContentType}
-							}
+					await call.RequestStream.WriteAsync(
+						new AppendReq {
+							ProposedMessage = new AppendReq.Types.ProposedMessage {
+								Id             = e.EventId.ToDto(),
+								Data           = ByteString.CopyFrom(e.Data.Span),
+								CustomMetadata = ByteString.CopyFrom(e.Metadata.Span),
+								Metadata = {
+									{ Constants.Metadata.Type, e.Type },
+									{ Constants.Metadata.ContentType, e.ContentType }
+								}
+							},
 						}
-					}).ConfigureAwait(false);
+					).ConfigureAwait(false);
 				}
-			} finally {
+
 				await call.RequestStream.CompleteAsync().ConfigureAwait(false);
+			} catch (RpcException) {
+				// Do nothing so that RpcExceptions propagate to the call.ResponseAsync and be translated by the TypedInterceptor
+			}
 
-				var response = await call.ResponseAsync.ConfigureAwait(false);
-
-				if (response.Success != null) {
-					writeResult = new SuccessResult(response.Success.CurrentRevisionOptionCase ==
-					                                AppendResp.Types.Success.CurrentRevisionOptionOneofCase.NoStream
-							? StreamRevision.None
-							: new StreamRevision(response.Success.CurrentRevision),
-						response.Success.PositionOptionCase == AppendResp.Types.Success.PositionOptionOneofCase.Position
-							? new Position(response.Success.Position.CommitPosition,
-								response.Success.Position.PreparePosition)
-							: default);
-					_log.LogDebug("Append to stream succeeded - {streamName}@{logPosition}/{nextExpectedVersion}.",
-						header.Options.StreamIdentifier, writeResult.LogPosition, writeResult.NextExpectedStreamRevision);
-				} else {
-					if (response.WrongExpectedVersion != null) {
-						var actualStreamRevision = response.WrongExpectedVersion.CurrentRevisionOptionCase switch {
-							AppendResp.Types.WrongExpectedVersion.CurrentRevisionOptionOneofCase.CurrentNoStream =>
+			var response = await call.ResponseAsync.ConfigureAwait(false);
+
+			if (response.Success != null) {
+				writeResult = new SuccessResult(
```

Comment:
Can we improve the readability of this?
Perhaps create the different elements required to create the `SucessResult` before and then just pass them in.

Reply:
Done 👍